html/captive-portal/lib/captiveportal/PacketFence/DynamicRouting/Module/Authentication/SMS.pm

```diff
@@ -91,10 +91,14 @@ Prompt fields with source specific SMS carriers
 sub prompt_fields {
     my ($self) = @_;
 
-    my @carriers = map { { label => $_->{name}, value => $_->{id} } } @{sms_carrier_view_all($self->source)};
-    $self->SUPER::prompt_fields({
-        sms_carriers => \@carriers, 
-    });
+    unless ( $self->source->type eq "Twilio" ) {
```

Comment:
Instead of Twilio , shouldn't we use an extra parameter like use_api and something like use_email because i think we probably have in the future more sms sources that use an API.